linker/src/bus.rs

```diff
@@ -27,7 +27,7 @@ pub struct BusLinker {
     /// arguments for `bus_multi_linker`
     bus_multi_linker_args: ArrayLiteral,
     /// interaction id map for each link.to
-    interaction_id_map: HashMap<LinkTo, u32>,
+    interaction_id_map: BTreeMap<(Operation, Location), u32>,
```

Comment:
iirc `LinkTo` has the location and operation, so it seems clearer to use that as you started with!